### ISSUE TITLE
Config displayFilename path.js error

### ISSUE THREAD

**Issue by ManUtopiK** [klauscfhq/signale#5]
**Describe the bug**
Adding `displayFilename: true` config throw this error:
'"The "path" argument must be of type string"'
Stack error:
```error
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string
at assertPath (path.js:39:11)
at Object.basename (path.js:1300:5)
at stack.map.x (/MYAPP/node_modules/signale/signale.js:63:41)
at Array.map (<anonymous>)
at Signale.get filename [as filename] (MYAPP/node_modules/signale/signale.js:63:27)
at Signale._formatFilename (MYAPP/node_modules/signale/signale.js:91:21)
at Signale._meta (MYAPP/node_modules/signale/signale.js:115:22)
at Signale._buildSignale (MYAPP/node_modules/signale/signale.js:142:26)
at Signale._logger (MYAPP/node_modules/signale/signale.js:79:20)
at Object.<anonymous> (MYAPP/src/hooks/logger.js:17:12)
```
**To Reproduce**
Install [feathers.js](https://github.com/feathersjs/feathers).
Replace `winston` by `signale` in `sr/index.js` and `scr/hooks/logger.js`. It's work.
Add this config after `const logger = require('signale');`:
```js
logger.config({
  displayFilename: true
});
```
💥  Boom! 💥 

**Technical Info (please complete the following information)**
 - OS: OSX 10.13.1 (17B1003)
 - Signale Version: ^1.0.1
 - Node.js Version: v9.11.1

**Comment by rjoydip** [comment]
@ManUtopiK I have checked but didn't get an issue. Replace Winston by signale in `src/index.js` and `scr/hooks/logger.js` along with `logger.config({ displayFilename: true });` It's working fine here.

**Comment by aguilera51284** [comment]
i have the same error with feathersjs

**Technical Info (please complete the following information)**

    OS: Ubuntu 18.04
    Signale Version: ^1.0.1
    Node.js Version: v9.11.1

**Comment by jordrake** [comment]
After much digging. I have managed to recreate this in a small snippet:

```js
const signale = require('signale');
signale.config({
  displayFilename: true
});

Promise.resolve().then(() => {
  console.log(new Error());
  signale.error('test');
});
```

It is caused by the presence of `<anonymous>` in the stack trace:
```
Error
    at Promise.resolve.then (/mnt/d/dev/signale/test.js:8:15)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
    at Function.Module.runMain (module.js:678:11)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
```
It has a callsite but has callsite.getFileName() returns null resulting in path.basename rejecting it causing this error.

I can provide a fix.

**Comment by klauscfhq** [comment]
I was able to reproduce the same issue on versions `8.9.1` and `9.11.1`, but not on version `v10.2.1`. The problem as @jordrake described has to do with `path.basename` and the fact that it produces the `TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string` error when null is passed as argument instead of a valid string path. The issue will be fixed once #30 is merged : )
